packages/opds-web-client/src/OPDSDataAdapter.ts

```diff
@@ -173,17 +173,21 @@ export function entryToBook(entry: OPDSEntry, feedUrl: string): BookData {
   };
 }
 
-function entryToLink(entry: OPDSEntry, feedUrl: string): LinkData {
-  let href: string;
+function entryToLink(entry: OPDSEntry, feedUrl: string): LinkData | null {
   let links = entry.links;
   if (links.length > 0) {
-    href = resolve(feedUrl, links[0].href);
+    const href = resolve(feedUrl, links[0].href);
+    return {
+      id: entry.id,
+      text: entry.title,
+      url: href
+    };
   }
-  return <LinkData>{
-    id: entry.id,
-    text: entry.title,
-    url: href
-  };
+  console.error(
```

Comment:
I added this. It used to return an object an undefined `href`, but that didn't seem correct. Let me know what you think here @EdwinGuzman

Reply:
I can't say I know for sure if this is right (as in, an acceptable fallback for the data structure).

Reply:
Yea in the past, it would have returned an object with an undefined `href`, so maybe I should just preserve that functionality I suppose for now?

Reply:
Yeah, I think for now just to be safe until we know for sure later.